staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -72,31 +71,46 @@ func init() {
 	dbMetricsMonitors = make(map[string]struct{})
 }
 
-func newETCD3HealthCheck(c storagebackend.Config) (func() error, error) {
+func newETCD3HealthCheck(c storagebackend.Config) (func() error, func(), error) {
```

Comment:
instead of returning a cleanup function, we could pass a shutdown signal (`<-chan struct{}`) and destroy the client once that chan has been closed. The channel could be taken from a lifecycle signal.